server/src/main/java/com/cloud/projects/ProjectManagerImpl.java

```diff
@@ -727,10 +734,18 @@ public Project updateProject(final long projectId, final String displayText, fin
         Transaction.execute(new TransactionCallbackWithExceptionNoReturn<ResourceAllocationException>() {
             @Override
             public void doInTransactionWithoutResult(TransactionStatus status) throws ResourceAllocationException {
-                if (displayText != null) {
+                if (displayText != null && name != null) {
+                    project.setDisplayText(displayText);
+                    project.setName(name);
+                    _projectDao.update(projectId, project);
+                } else if (name !=null) {
+                    project.setName(name);
+                    _projectDao.update(projectId, project);
+                } else {
                     project.setDisplayText(displayText);
                     _projectDao.update(projectId, project);
                 }
+
```

Comment:
this code is a repetition of l 659-669, please create a method

Reply:
fixed